plugins/PostProcessingPlugin/scripts/DisplayInfoOnLCD.py

```diff
@@ -217,18 +284,29 @@ def execute(self, data):
                             max_layer = str(int(max_layer) - 1)
                     if line.startswith(";LAYER:"):
                         if self.getSettingValueByKey("maxlayer"):
-                            display_text = display_text + " of " + max_layer
+                            display_text += " of " + max_layer
+                            m118_text += " of " + max_layer
                             if not self.getSettingValueByKey("scroll"):
-                                display_text = display_text + " " + file_name
+                                display_text += " " + file_name
+                                m118_text += " " + file_name
                         else:
                             if not self.getSettingValueByKey("scroll"):
-                                display_text = display_text + " " + file_name + "!"
+                                display_text += " " + file_name + "!"
+                                m118_text += " " + file_name + "!"
                             else:
-                                display_text = display_text + "!"
+                                display_text += "!"
+                                m118_text += "!"
                         line_index = lines.index(line)
-                        lines.insert(line_index + 1, display_text)
+                        if add_m117_line:
+                            lines.insert(line_index + 1, display_text)
                         if add_m118_line:
-                            lines.insert(line_index + 2, str(display_text.replace("M117", "M118", 1)))
+                            if add_m118_a1 and not add_m118_p0:
+                                m118_str = m118_text.replace("M118 ","M118 A1 ")
+                            if add_m118_p0 and not add_m118_a1:
+                                m118_str = m118_text.replace("M118 ","M118 P0 ")
+                            if add_m118_p0 and add_m118_a1:
+                                m118_str = m118_text.replace("M118 ","M118 A1 P0 ")
```

Comment:
elif variation, first checks for both being true, otherwise goes for the other conditions.
```suggestion
                            if add_m118_p0 and add_m118_a1:
                                m118_str = m118_text.replace("M118 ","M118 A1 P0 ")
                            elif add_m118_p0:
                                m118_str = m118_text.replace("M118 ","M118 P0 ")
                            elif add_m118_a1:
                                m118_str = m118_text.replace("M118 ","M118 A1 ")
```